File: dev/test_data/main/python/expected/aas_core_meta.v3/input/snippets/Verification/extension_names_are_unique.py
def extension_names_are_unique(extensions: Iterable[aas_types.Extension]) -> bool:
    """
    Check that all :py:attr:`.types.Extension.name` are unique
    among :paramref:`extensions`.
    """
    name_set = set()  # type: Set[str]
    for extension in extensions:
        if extension.name in name_set:
            return False

        name_set.add(extension.name)

    return True

File: dev/test_data/main/csharp/expected/aas_core_meta.v3/input/snippets/Verification/extension_names_are_unique.cs
/// <summary>
/// Check that all <see cref="Aas.IExtension.Name" /> are unique among
/// <paramref name="extensions" />.
/// </summary>
public static bool ExtensionNamesAreUnique(
    IEnumerable<Aas.IExtension> extensions
)
{
    var nameSet = new HashSet<string>();
    foreach (var extension in extensions)
    {
        if (nameSet.Contains(extension.Name))
        {
            return false;
        }
        nameSet.Add(extension.Name);
    }
    return true;
}
